.github/workflows/Dev-Promotion.yml

```
@@ -0,0 +1,100 @@
+---
+name: Dev promotion
+
+on: 
+#  push:
+#    branches: 
+#      - master
+#    paths-ignore:
+#      - '.github/workflows/**'
+#      - 'test-script'
+#      - '**/pom.xml' 
+  workflow_dispatch:
+
+env:
+  IMAGE_BASE_NAME: ceres-test
+  HELM_CHART: ceres
+  HELM_CHART_FOLDER: helm-ceres/cherts/ceres
+  CLUSTER_NAME: primary-cluster
+  REGISTRY: gcr.io
+
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.setVersion.outputs.version }}
+    steps:
+    ### Seting up runner env ###
+      - uses: actions/checkout@v3
+
+      - uses: ./.github/actions/GCPSetup
+        with:
+          GCPCreds: ${{ secrets.GCP_SA_JSON }} 
+
+      - name: 'Maven settings'
+        run: 'gsutil cp gs://salus-mavenrepository/m2-settings.xml ./.mvn/settings.xml'
+
+      - name: Docker login
+        run: gcloud auth configure-docker ${{ env.REGISTRY }} -q
+
+      - name: .m2 forder [*** fix it please ***]
+        run: sed -i '/<localRepository>/d' ./.mvn/settings.xml
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          java-version: '14'
+          distribution: 'adopt-hotspot'
+          cache: maven
+
+      - name: Set version
+        id: setVersion
+        shell: bash
+        run: |
+          POM_VERSION=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+          RUN_ID=${{ github.run_number }}
+          COMMIT_SHA=${{ github.sha }}
+          VERSION=${POM_VERSION}-${RUN_ID}-${COMMIT_SHA}
```

[Semver says](https://semver.org/#spec-item-10) build metadata is appended after a plus sign, so
```suggestion
          VERSION=${POM_VERSION}-${RUN_ID}+${COMMIT_SHA}
```
The run id here would be considered a "pre-release version", which is fine, I think. That figures into precedence order when comparing versions, but the commit sha is definitely build metadata that shouldn't be compared.